keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -795,79 +840,149 @@ enhancement:
 * **Are there any tests for feature enablement/disablement?**
   The e2e framework does not currently support enabling and disabling feature
   gates. However, unit tests in each component dealing with managing data created
-  with and without the feature are necessary. At the very least, think about
-  conversion tests if API types are being modified.
+  with and without the feature are necessary and will be added before
+  before the transition to beta.
 
 ### Rollout, Upgrade and Rollback Planning
 
-Will be added before the transition to beta.
-
 * **How can a rollout fail? Can it impact already running workloads?**
 
+A rollout happens in at least two phases:
+1. Updating the cluster so that the `CSIStorageCapacity` API is enabled in the apiserver
+   and the kube-scheduler uses that information *for drivers which have opted into this*.
+2. CSI driver installations get updated such that they produce `CSIStorageCapacity` objects
+   and enable usage of those objects in their `CSIDriver` object.
+
+If for whatever reason the first phase fails, the feature can be left
+disabled and the second phase can be skipped. The cluster the operates
+as before.
+
+If the second phase fails because a driver malfunctions or overloads
+the apiserver, then it can be rolled back and scheduling again happens
+without using storage capacity information.
+
+In none of these cases are running workloads affected.
+
 * **What specific metrics should inform a rollback?**
 
+Increased number of pods that are not getting scheduled and apiserver
+load.
+
 * **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
 
+No.
+
 * **Is the rollout accompanied by any deprecations and/or removals of features,
   APIs, fields of API types, flags, etc.?**
 
-### Monitoring requirements
+No.
 
-Will be added before the transition to beta.
+### Monitoring requirements
 
 * **How can an operator determine if the feature is in use by workloads?**
 
+The feature is used if all of the following points apply:
+- `CSIDriver.spec.storageCapacity` is true for a CSI driver.
+- There are storage classes for that driver.
+- Volumes are using those storage classes.
+
 * **What are the SLIs (Service Level Indicators) an operator can use to
   determine the health of the service?**
 
+Pod status of the CSI driver deployment, existence of
+`CSIStorageCapacity` objects and ultimately the rate at which volumes
+are being created by the driver. The latter is exposed through the
+existing CSI sidecar metrics support.
+
 * **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
 
+This depends on the workload and CSI driver. A very low rate of
+provisioned volumes per second and few total number of volumes overall
+may be fine for some workloads (long-running applications which use
+specialize storage like PMEM) while others may need a much higher rate
+(adding a local LVM scratch volume to every pod in the system).
```

Comment:
I think we don't yet know what exactly it should be - and that sounds fine for Beta. I would remove it for now and leave a TODO for GA (this is to ambiguous at this point).

Reply:
Would a reasonable SLO be to match what provisioning achieves today?

Reply:
That sounds fine to me. But then let's remove this paragraph :)

Reply:
ping - let's remove this paragraph

Reply:
Oh, so with "then" you meant "now" and not "when going to GA".

I'm not entirely sure what to remove - just the "need further discussion" or also "The goal is..."?

Reply:
Sorry - I meant:

"Let's remove now this part:

`This depends on the workload and CSI driver. A very low rate of
provisioned volumes per second and few total number of volumes overall
may be fine for some workloads (long-running applications which use
specialize storage like PMEM) while others may need a much higher rate
(adding a local LVM scratch volume to every pod in the system).
`

Reply:
Done.